All_home_works/hw1/task2.py

```diff
@@ -0,0 +1,30 @@
+"""
+Write down the function, which reads input line-by-line, and find maximum and minimum values.
+Function should return a tuple with the max and min values.
+For example for [1, 2, 3, 4, 5], function should return [1, 5]
+We guarantee, that file exists and contains line-delimited integers.
+To read file line-by-line you can use this snippet:
+with open("some_file.txt") as fi:
+    for line in fi:
+        ...
+"""
+from typing import Tuple
+
+
+def find_maximum_and_minimum(file_name: str) -> Tuple[int, int]:
+    with open(file_name) as data_file:
+        min_value = max_value = int(data_file.readline()[:-1])
+        for line in data_file.readlines():
```

Comment:
What does data_file.readlines() does?

Reply:
method readline() will read only one string in file and will return it
method readlines() will read all strings in file and will retun list => List[str]

Reply:
What problems we may have with `readlines()`?

Reply:
If a file is a huge, a lot of memory will be required to save it in a list. Because readlines() will read the file immediately, not line by line as readline()

Reply:
Can we avoid it?